sparrow-py/pysrc/sparrow_py/_timestream.py

```diff
@@ -175,8 +181,28 @@ def __add__(self, rhs: Union[Timestream, Literal]) -> Timestream:
         -------
         Timestream
             The Timestream resulting from `self + rhs`.
-        """
-        return Timestream._call("add", self, rhs)
+
+        Raises
+        ------
+        ValueError
+            If attempting to perform an addition that is not possible.
+        """
+        if isinstance(rhs, timedelta):
```

Comment:
This is a bit ugly. Basically, `timedelta` is represented as `days, seconds, microseconds`. We need to decide whether add days means "add 24*60*60 seconds" or "add 1 calendar day". Given it is kept in a granular form, and we said `timedelta(days=5)` I interpreted this as "add 5 calendar days".

Then the other problem is that that doesn't match the Arrow (or Sparrow) representation, which has separate "duration" and "interval" columns. Under the hood, a timedelta is almost a triple of `{ day: IntervalDaysColumn, seconds: DurationSeconds, microseconds: DurationMicroseconds }`. We could consider that as a generic representation, possibly with some optimizations when we know the column is zero. Short of that, we'll need to spend some time thinking about how to map the Python ("just timedelta") to Arrow (and Sparrow), and probably do some more work under the hood.

Reply:
Actually, it's unclear how that would even work if you wanted to add "1 month". How much is that? Is it Februray? Is it a leap year?

This is why we also have an `IntervalMonths` column type.

Reply:
It seems like we'd want to actually use `dateutil.relativedelta` for that (https://dateutil.readthedocs.io/en/stable/relativedelta.html).

So maybe the answer is:

1. `timedelta` is always an absolute (fixed) number of seconds.
2. `relativedelta` is used to capture a number of months / days / seconds to add.

But that is a bigger can of worms.

Reply:
I think given ^^^ that, this should be https://docs.python.org/3/library/datetime.html#datetime.timedelta.total_seconds

And then we can use relativedelta if we need other things.

Reply:
Ah okay, that makes sense. So if we do have a literal `timedelta` we can just do the following?
```
result = Timestream._call("add_time", rhs.total_seconds(), result, session=session)
```

Reply:
1. Need to convert to `seconds` first (and convert to int):

```
seconds = Timestream._call("seconds", int(rhs.total_seconds(), session=session)
return Timestream._call("add_time", seconds, self)
```